lib/ansible/playbook/role_include.py

```diff
@@ -88,9 +88,17 @@ def get_block_list(self, play=None, variable_manager=None, loader=None):
         actual_role = Role.load(ri, myplay, parent_role=self._parent_role, from_files=from_files,
                                 from_include=True, validate=self.rolespec_validate, public=self.public)
         actual_role._metadata.allow_duplicates = self.allow_duplicates
+        actual_role.static = self.statically_loaded
 
-        if self.statically_loaded or self.public:
+        if self.public:
             myplay.roles.append(actual_role)
+        elif self.public is None:
+            if actual_role.static:
+                if not C.DEFAULT_PRIVATE_ROLE_VARS:
+                    myplay.roles.append(actual_role)
+            else:
+                # NOTE C.DEFAULT_PRIVATE_ROLE_VARS does not have any effect on include_role, is that correct?
+                ...
```

Comment:
I think the intention was that the config setting would be used as the default for both include_role and static roles. I noticed it wasn't working with `include_role` in https://github.com/ansible/ansible/pull/81772#discussion_r1336503307 and thought it was fixed from a code review, but didn't test again. This seems like it could be:
```suggestion
+        if self.public or (self.public is None and not C.DEFAULT_PRIVATE_ROLE_VARS):
            myplay.roles.append(actual_role)
```

But I'm wondering if it would break backwards compatibility because `include_role` is private by default and falls back to `DEFAULT_PRIVATE_ROLE_VARS` which is the opposite, public by default? Maybe `DEFAULT_PRIVATE_ROLE_VARS` needs to default to a sentinel value too and have some different handling between static vs not.

Reply:
The sentinel value is what I was thinking too. I just committed it to see if all tests pass (testing just on `roles` and `include_import` locally to verify). We would need to document it and add missing tests if we decide to go with it.

Reply:
you would use the sentinel to handle the values the other way around, in this case we only use the setting IF the keyword is not set, not if the setting was assigned.